src/cedalion/sim/synthetic_artifact.py

```diff
@@ -0,0 +1,365 @@
+
+    Returns:
+        DataFrame with event timing data (columns onset_time, duration, trial_type,
+        value, channel).
+    """
+
+    if timing is None:
+        timing = pd.DataFrame(columns=TIMING_COLUMNS)
+
+    event_time = 0
+    total_time = time[-1].item() - time[0].item()
+
+    while event_time < total_time * perc_events:
+        onset_time = np.random.uniform(time[0], time[-1])
+        duration = np.random.uniform(min_dur, max_dur)
+        type = np.random.choice(types)
+        if onset_time + duration < time[-1]:
+            timing = add_event_timing([(onset_time, duration)], type, channels, timing)
+            event_time += duration
+    return timing
+
+
+def sliding_window(ts: cdt.NDTimeSeries, window_size, step_size):
+    """Generates overlapping windows from the input timeseries.
+
+    Args:
+        ts: fNIRS timeseries data (i.e. amp or OD).
+        window_size: Size of the window in s.
+        step_size: Step size in s.
+
+    Returns:
+        List of windows.
+    """
+
+    num_windows = int((ts["time"][-1].item() - window_size) // step_size + 1)
+    windows = [
+        ts.sel(time=slice(i * step_size, i * step_size + window_size))
+        for i in range(num_windows)
+    ]
+    return windows
+
+
+def calculate_amplitudes(windows):
+    """Calculates the amplitude (max-min) for each window."""
+
+    windows = windows.pint.dequantify()
+
+    amplitudes = [np.max(window) - np.min(window) for window in windows]
+    return np.array(amplitudes)
+
+
+def compute_alpha(ts: cdt.NDTimeSeries, window_size, step_size):
+    """Computes the Alpha parameter using the sliding window approach.
+
+    Args:
+        ts: fNIRS timeseries data (i.e. amp or OD).
+        window_size: Size of the window in s.
+        step_size: Step size in s.
+
+    Returns:
+        Median amplitude of the windows.
+    """
+
+    windows = sliding_window(ts, window_size, step_size)
+    amplitudes = calculate_amplitudes(windows)
+    return np.median(amplitudes)
+
+
+def add_artifact_direct(
+    ts: cdt.NDTimeSeries, timing: tuple[float, float], artifact_func, scale: float = 1.0
+):
+    """Add a single artifact to a timeseries with direct scaling (mainly for testing).
+
+    Args:
+        ts: timeseries data
+        timing: Tuple with onset time, duration of the artifact.
+        artifact_func: Artifact function. Artifact function must take args (time).
+        scale: scale for artifact.
+
+    Returns:
+        Timeseries data with added artifact.
+    """
+
+    ts_copy = ts.copy()
+    unit = ts_copy.pint.units if ts_copy.pint.units else 1
+
+    artifact = artifact_func(ts_copy.time, timing[0], timing[1])
+    ts_copy += artifact * scale * unit
+    return ts_copy
+
+
+def add_artifacts(
+    ts: cdt.NDTimeSeries,
+    timing: pd.DataFrame,
+    artifacts: dict[str, ArtifactFunction],
+    mode: str = "auto",
+    scale: float = 1.0,  # FIXME combine with mode
```

Comment:
I would suggest to combine these two parameters to `scale : float | None = None`.
If `scale=None` use automatic scaling, otherwise use the value for manual scaling.